src/sagemaker/serve/builder/model_builder.py

```diff
@@ -896,8 +913,11 @@ def build(  # pylint: disable=R0911
 
     def _build_validations(self):
         """Validations needed for model server overrides, or auto-detection or fallback"""
-        if self.mode == Mode.IN_PROCESS:
-            raise ValueError("IN_PROCESS mode is not supported yet!")
+        if self.mode == Mode.IN_PROCESS and self.model_server is not ModelServer.MMS:
+            raise ValueError(
+                "IN_PROCESS mode is not supported yet for model server. It is "
+                "supported for MMS/Transformers server in beta release."
```

Comment:
nit: `IN_PROCESS mode is only supported for MMS/Transformers server in beta release.`

Reply:
This is better wording, I will change it thank you!